target/min.c

```diff
@@ -493,15 +494,26 @@ static void rx_byte(struct min_context *self, uint8_t byte)
             }
             else {
                 self->rx_frame_seq = 0;
-                self->rx_frame_state = RECEIVING_LENGTH;
+                self->rx_frame_state = RECEIVING_LENGTH_MSB;
             }
             break;
         case RECEIVING_SEQ:
             self->rx_frame_seq = byte;
             crc32_step(&self->rx_checksum, byte);
-            self->rx_frame_state = RECEIVING_LENGTH;
+            self->rx_frame_state = RECEIVING_LENGTH_MSB;
             break;
-        case RECEIVING_LENGTH:
+        case RECEIVING_LENGTH_MSB:
+#if (MAX_PAYLOAD > UINT8_MAX)
+            self->rx_frame_length = ((uint16_t)byte) << 8;
+            crc32_step(&self->rx_checksum, byte);
+            self->rx_frame_state = RECEIVING_LENGTH_LSB;
+            break;
+#else
+            // MSB == LSB if value is 8-bit
+            self->rx_frame_state = RECEIVING_LENGTH_LSB;
```

Comment:
This is not required at all, but maybe useful for debugging purposes (so the state variable always represents the actual state).